next.config.js

```diff
@@ -66,4 +66,65 @@ module.exports = {
       },
     ];
   },
+  async headers() {
+    const securityPolicies = {
+      'default-src': ["'self'"],
+      'script-src': [
+        "'self'",
+        // next or one package is actually use `eval` or `new Function`
+        "'unsafe-eval'",
+        'https://stats.data.gouv.fr',
+      ],
+      // Recommended on https://csp-evaluator.withgoogle.com/
+      'object-src': ["'none'"],
+      'style-src': [
+        "'self'",
+        // nextjs inlines CSS
+        "'unsafe-inline'",
+      ],
+      'connect-src': ["'self'", 'https://stats.data.gouv.fr'],
+      'img-src': [
+        "'self'",
+        // next/image component inlines images with `src="data:..."`
+        'data:',
+        // all domains of images in contributors list
+        'https://media-exp1.licdn.com',
+        'https://www.gravatar.com',
+        'https://avatars.githubusercontent.com',
+        'https://pbs.twimg.com',
+        'https://sibyll.in',
+      ],
+      'frame-src': ["'self'", 'https://stats.data.gouv.fr'],
+    };
+
+    return [
+      {
+        source: '/:path*',
+        headers: [
+          {
+            key: 'Content-Security-Policy',
+            value: Object.entries(securityPolicies)
+              .map(([securityKey, values]) => `${securityKey} ${values.join(' ')}`)
+              .join('; '),
+          },
+          {
+            key: 'X-Frame-Options',
+            value: 'DENY',
```

Comment:
I find it a bit sad that we now by default block embedding the contribution tool in an iframe 🙂

Reply:
Well unfortunately, the other possible value is `SAMEORIGIN` so it won't help.

We can remove it later on I guess, but for now does anybody tried to put ota.org in an iframe?

Reply:
Ok so let's keep it that way to maximize our score on dashlord